src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -213,8 +213,9 @@ public Index checkIfRoomPresent(Integer roomNumber) {
     public void updateRoomListWhenPatientsChanges(Patient patientToEdit, Patient editedPatient) {
         ObservableList<Room> roomObservableList = this.roomList.getRoomObservableList();
         for (int i = 0; i < roomObservableList.size(); i++) {
-            if (isPatientAssignedToRoom(patientToEdit.getName())
-                && roomObservableList.get(i).getPatient().isSamePatient(patientToEdit)) {
+            Patient patient = roomObservableList.get(i).getPatient();
+            if (isPatientAssignedToRoom(patientToEdit.getName()) && patient != null
```

Comment:
Comparing patient to null is not very elegant, maybe check if roomObservableList.get(i).isOccupied() instead? (presumably isOccupied = true means patient != null)

Reply:
oki, i will do that next time HAHHAHA, didnt see